server/testutil/app/app.go

```diff
@@ -48,10 +50,29 @@ func Run(t *testing.T, commandPath string, commandArgs []string, configFilePath
 
 }
 
+func createMemoryBackedDB(t testing.TB) *os.File {
+	if _, err := os.Stat("/dev/shm"); errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+	f, err := os.CreateTemp("/dev/shm", "buildbuddy.db")
```

Comment:
bazel is probably provisioning a `/dev/shm` for each test sandbox when this is run locally, but it might be a bit safer to include a `*` in the filename to prevent clobbering

```suggestion
	f, err := os.CreateTemp("/dev/shm", "buildbuddy.db")
```

```suggestion
	f, err := os.CreateTemp("/dev/shm", "buildbuddy-test-*.db")
```